src/ptrace_control.rs

```diff
@@ -2,58 +2,59 @@ use std::ptr;
 use nix::sys::signal::Signal;
 use nix::sys::ptrace::{ptrace, ptrace_setoptions};
 use nix::sys::ptrace::ptrace::*;
-use nix::libc::{pid_t, c_void, c_long};
+use nix::libc::{c_void, c_long};
+use nix::unistd::Pid;
 use nix::Result;
 use std::mem;
 
 const RIP: u8 = 128;
 
 
-pub fn trace_children(pid: pid_t) -> Result<()> {
+pub fn trace_children(pid: Pid) -> Result<()> {
     //TODO need to check support.
     let options: PtraceOptions = PTRACE_O_TRACESYSGOOD | PTRACE_O_TRACEEXEC | PTRACE_O_TRACEEXIT |
         PTRACE_O_TRACECLONE | PTRACE_O_TRACEFORK | PTRACE_O_TRACEVFORK;
     ptrace_setoptions(pid, options)
 }
 
-pub fn detach_child(pid: pid_t) -> Result<c_long> {
+pub fn detach_child(pid: Pid) -> Result<c_long> {
     ptrace(PTRACE_DETACH, pid, ptr::null_mut(), ptr::null_mut())
 }
 
-pub fn continue_exec(pid: pid_t, sig: Option<Signal>) -> Result<c_long> {
+pub fn continue_exec(pid: Pid, sig: Option<Signal>) -> Result<c_long> {
```

Comment:
So all instances of continue_exec can be replaced by `sys::ptrace::cont` in nix. NIx has a bunch of new utility functions that fulfil a lot of what ptrace_control did for me